test/e2e/framework/framework.go

```diff
@@ -0,0 +1,327 @@
+	t.Helper()
+	actual, err := ioutil.ReadFile(output)
+	if err != nil {
+		t.Fatalf("failed to read testdata file: %v", err)
+	}
+	if os.Getenv("UPDATE") != "" {
+		if err := os.MkdirAll(filepath.Dir(golden), 0755); err != nil {
+			t.Fatalf("failed to create fixture directory: %v", err)
+		}
+		if err := ioutil.WriteFile(golden, actual, 0644); err != nil {
+			t.Fatalf("failed to write updated fixture: %v", err)
+		}
+	}
+	expected, err := ioutil.ReadFile(golden)
+	if err != nil {
+		t.Fatalf("failed to read testdata file: %v", err)
+	}
+
+	diff := difflib.UnifiedDiff{
+		A:        difflib.SplitLines(string(expected)),
+		B:        difflib.SplitLines(string(actual)),
+		FromFile: golden,
+		ToFile:   output,
+		Context:  3,
+	}
+	diffStr, err := difflib.GetUnifiedDiffString(diff)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if diffStr != "" {
+		t.Errorf("got diff between expected and actual result: \n%s\n\nIf this is expected, re-run the test with `UPDATE=true go test ./...` to update the fixtures.", diffStr)
+	}
+}
+
+// CiOperatorCommand returns the basic ci-operator command and artifact dir. Add args and env as necessary.
+func CiOperatorCommand(t *testing.T) (*exec.Cmd, string) {
+	t.Helper()
+	ctx := context.Background()
+	if deadline, ok := t.Deadline(); ok {
+		c, cancel := context.WithDeadline(ctx, deadline.Add(-10*time.Second))
+		ctx = c
+		t.Cleanup(cancel) // this does not really matter but govet is upset
+	}
+	var artifactDir string
+	if dir, set := os.LookupEnv("ARTIFACT_DIR"); set {
+		artifactDir = filepath.Join(dir, strings.NewReplacer("/", "_", "\\", "_", ":", "_").Replace(t.Name()))
+		if err := os.MkdirAll(artifactDir, 0755); err != nil {
+			t.Fatalf("could not create artifact dir for ci-operator: %v", err)
+		}
+	} else {
+		artifactDir = t.TempDir()
+	}
+	t.Cleanup(func() {
+		if walkErr := filepath.Walk(artifactDir, func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			// if we do not mangle these file-names, the jUnit spyglass lens
+			// will show (somtimes expected) failures in these files from delegated
+			// ci-operator runs in the overview, which is confusing
+			if strings.HasPrefix(info.Name(), "junit") {
+				if err := os.Rename(path, strings.ReplaceAll(path, "/junit", "/_junit")); err != nil {
+					t.Logf("failed to mangle jUnit filename for %s: %v", path, err)
+				}
+			}
+			return nil
+		}); walkErr != nil {
+			t.Errorf("failed to walk fixture tree for comparison: %v", walkErr)
+		}
+	})
+	cmd := exec.CommandContext(ctx, "ci-operator",
+		"--input-hash="+strconv.Itoa(rand.Int()), // we need unique namespaces
+		"--artifact-dir="+artifactDir,
+	)
+	cmd.Env = append(cmd.Env, KubernetesClientEnv(t)...)
+	return cmd, artifactDir
+}
+
+func Execute(t *testing.T, cmd *exec.Cmd, testDone <-chan struct{}, cleanupDone chan<- struct{}) ([]byte, error) {
+	t.Helper()
+	verbose := os.Getenv("VERBOSE") == "true"
+	if verbose {
+		t.Logf("running: %v", cmd.Args)
```

Comment:
go itself will hide everything you log via `t.Log` unless the test fails or you run the tests with `-v` so IMHO we don't need another verbosity configuration layer